### DISCUSSION TITLE
Request - Suspend the physics of objects being overlapped UPBGE 2.5

### DISCUSSION THREAD

**Discussion by LucasSnSouza** [UPBGE/upbge#1680] [Ideas/Features for UPBGE 0.2.5+]
Hi, I'm on a game project and one of the things I'm trying to solve for a while in my game, for performance. the game consists of building your own ships, in a system of blocks. and one of the performance problems that I identified would be, when an object is completely covered by the other blocks, I can hide this block from the Rasterization, but its physics is not disabled. as it would be a spaceship, this object will move this keeping the object always active, not sleeping, I'm not aware of anything that helps this in the current UPBGE 2.5. and I'm also not sure if I make this request or question here, but it would be very useful if possible. than Object Culling. can also suspend the physics of objects that are superimposed by others.

**Comment by HolandonHdunto** [comment]
![Fisica](https://user-images.githubusercontent.com/89164699/157447660-84eee08a-8f84-417e-bdc7-07431272b685.gif)

**Comment by LucasSnSouza** [comment]
Hi, I know this function until I made a code with raycast so that it suspends itself physically. however this method does not suspend the physics when the object is being superimposed. My doubt and idea is to make a Physics Cut when the object cannot be seen directly by the camera, because it is superimposed by another, in the same way that Camera Culling works for Rendering, but doing the same for the physics of the deleted objects. Sorry for the way of writing, English is not my native language.

**Reply by HolandonHdunto** [reply]
Well, it shouldn't be possible to do that, but it would probably be better to use another way of playing the game, as occlusion is perhaps more problematic than a complete object in a single mesh.
But there is a way to suspend physics, using distance.
![fisica](https://user-images.githubusercontent.com/89164699/158667246-126f51c4-3d75-4556-8a80-451ecc4d4c47.PNG)
